ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -963,6 +1001,176 @@ def setup_jmt(self, curpt, target_velocity, accel_ratio=1.0,
+            if accel_rate < 0.0:
+                final_accel = accel_rate - acc_diff
+                duration = rospy.get_time() - timer_start
+                rospy.logdebug("greatest accel_rate of {3.2f} to accelerate with max_jerk"
+                        "={:3.3f} from v={:3.3f}, a={:3.3f} to v={:3.3f} in "
+                        "dist {:3.3f}m in time {:3.3f}s, took {:3.4f}s to calc"
+                        .format(final_accel, jerk, curpt.get_v(), curpt.get_a(), self.default_velocity,
+                                a_dist, T, duration))
+                return(final_accel,a_dist)
+
+            end = [curpt.JMTD.S + a_dist, self.default_velocity, 0.0]
+            T = get_accel_time(a_dist, curpt.JMTD.V, self.default_velocity)
+            rospy.logdebug("Test accel rate of {:3.2f} going from v={:3.3f}, a={:3.3f}"
+                           " to v={:3.3f} in dist {:3.3f} m in time {:3.3f} s"
+                           .format(accel_rate, curpt.get_v(), curpt.get_a(), self.default_velocity,
+                                a_dist, T))
+            jmt = JMT(start, end, T)
+            jerk = jmt.get_j_at(0.1)
+            acc = jmt.get_a_at(T - 0.5)
+
+            rospy.logdebug("Using accel = {:3.2f}, found initial jerk of {:3.2f} with a_dist of {:3.2f}"
+                       " and final acc of {:3.2f} with a time of {:3.2f}"
+                       .format(accel_rate, jerk, a_dist, acc ,T))
+    
+            if too_short is True:
+                # looking for first instance that matches
+                if jerk < max_jerk:
+                    final_accel = accel_rate
+                    optimized = True
+            else:
+                # looking for first instance that fails
+                if jerk  > max_jerk:
+                    final_accel = accel_rate - acc_diff
+                    a_dist = get_accel_distance(curpt.get_v(), self.default_velocity, final_accel,
+                                    curpt.get_a())
+                    jerk = old_jerk
+                    optimized = True
+ 
+            if counter > 30:
+                rospy.logwarn("counter is {} in get_max_accel - bail with acc={:3.2f}!"
+                              .format(counter, accel_rate))
+                final_accel = accel_rate
+                optimized = True
+        
+        duration = rospy.get_time() - timer_start
+        rospy.logdebug("greatest accel_rate of {:3.2f} to accelerate with max_jerk"
+                       "={:3.3f} from v={:3.3f}, a={:3.3f} to v={:3.3f} in "
+                       "dist {:3.3f}m in time {:3.3f}s, took {:3.4f}s to calc"
+                       .format(final_accel, jerk, curpt.get_v(), curpt.get_a(), self.default_velocity,
+                               a_dist, T, duration))
+        return (final_accel, a_dist)
+
+    def get_speedup_time(self, start, end):
+        # Use JMT to figure out proper time for deceleration where 
+        # curve does not wobbble below min_moving _velocity
```

Comment:
Change comment to "proper time for acceleration where curve does not wobble above max velocity"